File: search-insert-position.py
from typing import List
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        if target in nums:
            return nums.index(target)
        nums_len = len(nums)
        for i in range(0, nums_len):
            if nums[i]>=target:
                return i
        return nums_len
if __name__ == '__main__':
    s = Solution()
    print(s.searchInsert([1,3,5,6], 5))
    print(s.searchInsert([1,3,5,6], 2))
    print(s.searchInsert([1,3,5,6], 7))
    print(s.searchInsert([1,3,5,6], 6))
    print(s.searchInsert([1,3,5,6], 0))
    print(s.searchInsert([1,3,5,6], 5))

File: search-insert-position.go
package main

import "fmt"

func searchInsert(nums []int, target int) int {
	for i, num := range nums {
		if target <= num {
			return i
		}
	}
	return len(nums)
}
func main() {
	var test = []int{1, 3, 5, 6}
	fmt.Printf("%d\n", searchInsert(test, 5))
	fmt.Printf("%d\n", searchInsert(test, 2))
	fmt.Printf("%d\n", searchInsert(test, 7))
	fmt.Printf("%d\n", searchInsert(test, 6))
	fmt.Printf("%d\n", searchInsert(test, 0))
	fmt.Printf("%d\n", searchInsert(test, 5))
}
